### ISSUE TITLE
Justification broken with possibly-hyphenated ligatures

### ISSUE THREAD

**Issue by simoncozens** [simoncozens/sile#223]
Looks like SILE is using the wrong length to compute the width of ligated words.

```TeX
\begin[papersize=a4, class=book]{document}
\set[parameter=document.parindent,value=0pt]
\set[parameter=current.parindent,value=0pt]
\script[src=packages/frametricks]
\showframe
\define[command=testdoc]{Or la nativite de Jesus Christ estoit telle. Car comme Marie mere diceluy fut espousee a Joseph : devant 
quilz convinssent/ elle fut trouvee enceincte du sainct esprit. Et Joseph son mary/ estant juste/ & ne
voulant point la diffamer/ la voulut secretement laisser. Mais comme iceluy 
}

\font[filename=/usr/local/texlive/2015/texmf-dist/fonts/opentype/public/ebgaramond/EBGaramond12-Regular.otf, size=9pt, features="+cv01"]

\nohyphenation{\testdoc}

\testdoc

\end{document}
```

![screen shot 2015-11-28 at 22 27 32](https://cloud.githubusercontent.com/assets/106728/11452105/3daf8f00-961f-11e5-916d-25e5ebe3e35f.png)

**Comment by simoncozens** [comment]
And here's my explanation of this bug:

When SILE is asked to prepare hyphenation points for a paragraph, it measures two copies of each word, one unhyphenated and one hyphenated. When it measures "laisser", it will measure the unhyphenated width and also the hyphenated width "lais" + "ser". But "laisser" has a ligature which makes it *shorter* than the component pieces "lais" + "ser". SILE feeds the widths of the component pieces to the justification engine, but it actually ends up using the ligated form, so it gets the wrong numbers back.

**Comment by simoncozens** [comment]
Seems like this is fixed after #255.
